### PR TITLE
Userscript metadata extraction

### PR DESCRIPTION
**Pull request by Cerbrus**
Our userscripts should have a `.meta.js` next to a `.user.js`, to reduce the load on systems hosting the scripts, as explained [in this SO answer](http://stackoverflow.com/a/38025376/1835379).

I've added a `extractUserScriptMeta.sh` script that grabs the metadata (requires a blank line after the metadata), and saves it in a `scriptname.meta.js` file.  
GreaseMonkey / TamperMonkey should automatically look for these files when checking if a script needs updating.

The `extractUserScriptMeta` could be called from `build.sh`.  
There are probably also more efficient ways to extract the metadata block. (review please?)

### CODE REVIEW

File: fire/built.meta.bat [hunk 1]
```diff
@@ -0,0 +1,16 @@
+@ECHO OFF
```
Comment (j-f1) [inline fire/built.meta.bat]: What is this file here for?

File: extractUserScriptMeta.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+# so the whole script doesn't have to be downloaded to check for updates.
+# Example use:
+# // @updateURL   https://raw.githubusercontent.com/Charcoal-SE/Userscripts/master/fire/fire.meta.js
+# // @downloadURL https://raw.githubusercontent.com/Charcoal-SE/Userscripts/master/fire/fire.user.js
+
+find ./ -name "*user.js" -type f | while read file; do
+  newName=${file/.user./.meta.}
+  echo "Processing file '$file' into '$newName'"
+  
+  sed -e '/./{H;$!d;}' -e 'x;/UserScript/!d;' $file > $newName # Copy only metadata,
```
Comment (ArtOfCode-) [inline extractUserScriptMeta.sh]: I'm gonna assume you know what you're doing with `sed`, because I sure don't :)

File: build.sh [hunk 3]
```diff
@@ -43,3 +43,7 @@ do
   fi
   cd ..
 done
+
+echo "Extracting userscript meta blocks."
+
+extractUserScriptMeta.sh
```
Comment (ArtOfCode-) [inline build.sh]: If this is in the same directory as build.sh, it'll need a `./` in front of it for the call to work. The file will need to be chmodded for the `x` bit, as well, which will require someone running not-Windows.

Comment (Cerbrus) [inline build.sh]: I've fixed the path already, yea. Completely forgot about that `x` bit...

**Pull request merged**